### PR TITLE
A redo of my chapter 5 demo

### PR DESCRIPTION
**Pull request by ssoegriemsingh**

### CODE REVIEW

File: 1732Sprint/3-Sheik/Chapter5/ClassDesign.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import classes.Knight;
+import classes.Paladin;
+import classes.Villager;
+
+import java.util.Arrays;
+import java.util.ArrayList;
+
+public class ClassDesign {
+
+    private static final String LINEBREAK = "\n";
```
Comment (njigalov) [inline 1732Sprint/3-Sheik/Chapter5/ClassDesign.java]: Nice.

File: 1732Sprint/3-Sheik/Chapter5/ClassDesign.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+    private ClassDesign() {
+        ArrayList<Character> charList = createCharacters();
+        displayCharacters(charList);
+    }
+
+    private ArrayList<Character> createCharacters() {
+        ArrayList<Character> charList = new ArrayList<>();
+
+        Knight k = new Knight();
```
Comment (njigalov) [inline 1732Sprint/3-Sheik/Chapter5/ClassDesign.java]: I know k is obvious here, but avoid single char variable names.

File: 1732Sprint/3-Sheik/Chapter5/classes/Character.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
+package classes;
+
+public abstract class Character {
+
+    public String name;
```
Comment (njigalov) [inline 1732Sprint/3-Sheik/Chapter5/classes/Character.java]: You sure this should be public? I see that I can create a Knight character and then give him the name, 'Bird', no?

**Comment by ssoegriemsingh** [comment]
Something went wrong, created a new pull request.

**Pull request closed without merging**